packages/plugin-page-view-tracking/src/plugin-campaign-tracker.ts

```diff
@@ -0,0 +1,53 @@
+import { AttributionBrowserOptions, Campaign } from '@amplitude/analytics-types';
+import { CampaignTracker } from '@amplitude/analytics-client-common';
+import { BASE_CAMPAIGN } from './constant';
+import { Storage } from '@amplitude/analytics-types';
+
+export class PluginCampaignTracker {
+  private _currentCampaign: Campaign = {
+    ...BASE_CAMPAIGN,
+  };
+  private _isNewCampaign = true;
+  ready: Promise<unknown>;
+  campaignTracker: CampaignTracker;
+
+  constructor(apiKey: string, storage: Storage<Campaign>, options: AttributionBrowserOptions) {
+    this.campaignTracker = new CampaignTracker(apiKey, {
```

Comment:
i'm worried that with this shared with browser bundled and plugin bundled feature, it will be harder to maintain since you want to make sure they one does not affect the other. while many code is shared today, i think they have diverged already and should be maintained separately

Reply:
We only use a few functions from `CampaignTracker`. For any additional functionality, we can stay in the plugin. I am feeling `CampaignTracker` is more like a tool which provides some basic campaign functions. 

If MA team needs more features or function, only plugins need to be updated. We might not want to modify or change `CampaignTracker` with new features.

Reply:
i feel like campaign parser is the generic one, while campaign tracker is the one that's different. the config alone does not match